.cursor/BUGBOT.md

```diff
@@ -1,120 +1,647 @@
-# Project Review Guidelines - Application SDK
+# Application SDK Code Review Guidelines
 
-## Core Principles
+## Review Mental Framework
 
-**Less Code is Better** - Prioritize simplicity, readability, and maintainability over cleverness. Every line of code is a liability that must be justified.
+Follow this systematic review process that mirrors how experienced developers think through code changes.
 
-## Critical Review Checklist
+### Phase 1: Immediate Safety Assessment
 
-### 🔍 Code Quality & Minimalism
+**Mental Question: "Could this cause immediate harm?"**
+Review for critical issues first - these take priority over everything else.
 
-- **Necessity Check**: Is this code absolutely necessary? Can existing functionality be reused?
-- **Single Responsibility**: Does each function/class do exactly one thing well?
-- **DRY Violations**: Any repeated logic that should be extracted into shared utilities?
-- **Cognitive Load**: Can a new developer understand this change with less cognitive load?
-- **Magic Numbers/Strings**: All constants properly defined and named in @application_sdk/constants.py
-- **Spell Check**: No typos in code, comments, docstrings, or documentation
-- **Exception Handling Standards**: All exception handling must follow these principles. See detailed guidelines in [exception-handling.mdc](@.cursor/rules/exception-handling.mdc).
+#### Security Vulnerabilities
 
-### 🏗️ Architecture & Design
+**Always flag immediately:**
 
-- **Module Boundaries**: Changes respect existing module responsibilities
-- **Error Handling**: Proper exception handling with meaningful error messages
-- **Resource Management**: Files, connections, and resources properly closed/released
-- **Async Patterns**: Proper async/await usage where applicable
-- **Configuration**: No hardcoded values; use proper configuration management
+- Hardcoded secrets, passwords, API keys, or tokens anywhere in code
+- SQL queries using string concatenation instead of parameters
+- User input directly included in system commands without sanitization
+- Missing authentication checks on protected operations
+- Sensitive data in log messages or error responses
+- Unsafe deserialization of user-provided data
 
-### 📝 Documentation Requirements
+**Python SDK specific security patterns:**
 
-- **Docstrings**: All public functions, classes, and modules have Google-style docstrings
-- **Type Hints**: All function parameters and return values are typed
-- **Complex Logic**: Non-obvious business logic has inline comments explaining "why"
+- Missing input validation in handler functions
+- Direct SQL execution without parameterization in `application_sdk/clients/sql.py`
```

Comment:
no-one actually writing sql queries in `clients/sql.py`, they actually write it inside a `queries/`